tests/e2e/utils/helpers.go

```diff
@@ -128,10 +128,11 @@ func HTTPGetNTimes(url string, n int) ([]byte, error) {
 
 // httpGet is a helper to make GET request call to url.
 func httpGet(url string, timeout time.Duration) ([]byte, error) {
-	resp, err := httpGetRaw(url, timeout) //nolint
+	resp, err := httpGetRaw(url, timeout)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
```

Comment:
scary shit. Nice catch.

Reply:
It was actually ok. Before, this was inside `extractBody` so it was being executed.

But......

1. Being inside `extractBody` meant that the linter would not see it so it was marked as error. This came from the `bodyclose` linter.
2. However, `bodyclose` is [not working with Go 1.18 yet](https://github.com/golangci/golangci-lint/issues/2649).
3. This means that the `//nolint` rule that was on that line was actually not doing anything, because no linter was finding error there (because `bodyclose` is not enabled until it's fixed for Go 1.18)
4. Because `//nolint` was not doing anything, another linter, `nolintlint` was throwing an error saying that there was a `//nolint` command that was necessary

Now, I could have just disabled `nolintlint`, but I figured that solving the problem at the root (and making it so we won't need `//nolint:bodyclose`) would just be easier.

PS: Did I say how much I ❤️ linters